src/plugins/editor/editor.tsx

```diff
@@ -76,14 +82,21 @@ export class UIPEditor extends UIPPluginPanel {
     return CodeJar(this.$code, UIPEditor.highlight, {tab: '\t'});
   }
 
+  public get isJsReadonly(): boolean {
+    return !this.model?.activeSnippet?.isolated && this.source !== 'html';
+  }
+
   /** @returns editor's content */
   public get value(): string {
     return this.editor.toString();
```

Comment:
Potentially dangereous  place if I understand correctly... 
It will indirectly make editable our area